ndcube/wcs/wcs_tools.py

```diff
@@ -0,0 +1,136 @@
+from numbers import Integral
+
+import numpy as np
+
+from astropy.wcs import WCS
+from astropy.wcs.wcsapi import SlicedLowLevelWCS
+from astropy.wcs.wcsapi.wrappers.base import BaseWCSWrapper
+from ndcube.wcs.wrappers import ResampledLowLevelWCS
+
+__all__ = ["unwrap_wcs_wrappers_to_fitswcs"]
+
+
+def unwrap_wcs_wrappers_to_fitswcs(wcs):
+    """Create FITS-WCS equivalent to WCS wrapper object.
+
+    Underlying WCS must be FITS-WCS.
+
+    Parameters
+    ----------
+    wcs: `BaseWCSWrapper`
+        The WCS Wrapper object. Base level WCS implementation must be FITS-WCS.
+        Current supported wrapper classes include `astropy.wcs.wcsapi.SlicedLowLevelWCS`
+        and `ndcube.wcs.wrappers.ResampledLowLevelWCS`.
+
+    Returns
+    -------
+    fitswcs: `astropy.wcs.WCS`
+        The equivalent FITS-WCS object.
+    """
+    low_level_wrapper = wcs.low_level_wcs
+    if isinstance(low_level_wrapper, WCS):
+        return low_level_wrapper
+    # Determine chain of wrappers down to the FITS-WCS.
+    wrapper_chain = [type(low_level_wrapper)]
+    while hasattr(low_level_wrapper, "_wcs"):
+        low_level_wrapper = low_level_wrapper._wcs.low_level_wcs
+        wrapper_chain.append(type(low_level_wrapper))
+    if not isinstance(low_level_wrapper, WCS):
+        raise TypeError(f"Base-level WCS must be type {type(WCS)}. Found: {type(low_level_wcs)}")
+    fitswcs = low_level_wrapper
+    for low_level_wrapper in wrapper_chain[::-1]:
+        if isinstance(low_level_wrapper, ResampledLowLevelWCS):
+            fitswcs = resample_fitswcs(fitswcs, low_level_wrapper.factor, low_level_wrapper.offset)
+        elif isinstance(low_level_wrapper, SlicedLowLevelWCS):
+            fitswcs = slice_fitswcs(fitswcs, low_level_wrapper._slices_pixel)
+        else:
+            raise TypeError("Unrecognized/unsupported WCS Wrapper type: {type(low_level_wrapper)}")
+     return fitswcs
+
+
+def slice_fitswcs(fitswcs, slice_items):
```

Comment:
You might find the `slice` and `sub` methods on `WCS` useful: https://docs.astropy.org/en/stable/api/astropy.wcs.WCS.html#astropy.wcs.WCS.slice